tests/framework/db/ActiveQueryTest.php

```diff
@@ -136,6 +136,26 @@ public function testInnerJoinWith()
         ], $result->joinWith);
     }
 
+    public function testBuildJoinWithRemoveDuplicateJoinByTableName()
+    {
+        $query = new ActiveQuery(Customer::className());
+        $query->innerJoinWith('orders')
+            ->joinWith('orders.orderItems');
```

Comment:
What about closure as a `on` statement?
https://www.yiiframework.com/doc/guide/2.0/en/db-active-record#relation-table-aliases

Reply:
Do you mean this case?
```php
$query = new ActiveQuery(Customer::className());
$query->innerJoinWith('orders')
    ->joinWith('orders', function($q) {
        $q->joinWith('orderItems');
    });
```
Works well.

Reply:
```php
$query->joinWith([
    'orders' => function ($q) {
        $q->from(['o' => Order::tableName()]);
    },
])
```
and
```php
$query->joinWith([
    'orders' => function ($q) use ($value) {
        $q->where(['key' => $value]);
    },
])
```

Reply:
I added tests for it to ActiveRecordTest. Why build failed after that? Failure in `FixtureControllerTest`: 
```
1) yiiunit\framework\console\controllers\FixtureControllerTest::testLoadActiveFixtureSequence
Failed asserting that two arrays are equal.
--- Expected
+++ Actual
@@ @@
 Array (
-    0 => 'yiiunit\data\console\controllers\fixtures\SecondIndependentActiveFixture'
-    1 => 'yiiunit\data\console\controllers\fixtures\FirstIndependentActiveFixture'
+    0 => 'yiiunit\data\console\controllers\fixtures\FirstIndependentActiveFixture'
+    1 => 'yiiunit\data\console\controllers\fixtures\SecondIndependentActiveFixture'
     2 => 'yiiunit\data\console\controll...ixture'
 )

/home/runner/work/yii2/yii2/tests/framework/console/controllers/FixtureControllerTest.php:244
/home/runner/work/yii2/yii2/vendor/phpunit/phpunit/phpunit:52
```

Reply:
That test was incorrect. I've fixed it.